src/trie.c

```diff
@@ -0,0 +1,92 @@
+/*
+   A trie data structure
+*/
+
+#include <stdlib.h>
+#include <assert.h>
+#include <math.h>
+#include "trie.h"
+#include "utils.h"
+
+/* See trie.h */
+trie_t *trie_new(char current) {
+    trie_t *t = calloc(1,sizeof(trie_t));
+
+    if (t == NULL) {
+        error("Could not allocate memory for trie_t");
+        return NULL;
+    }
+
+    t->current = current;
+
+    t->children = calloc(256,sizeof(trie_t*));
+    if (t->children == NULL) {
+        error("Could not allocate memory for t->children");
+        return NULL;
+    }
+
+    t->is_word = 0;
+    t->parent = NULL;
+
+    return t;
+}
+
+/* See trie.h */
+int trie_free(trie_t *t) {
+    assert( t != NULL);
+
+    for (int i=0; i<256; i++ ) {
+        if (t->children[i] !=NULL)
+            trie_free(t->children[i]);
+    }
+
+    free(t);
+    return EXIT_SUCCESS;
+}
+
+
+/* See trie.h */
+int trie_add_node(trie_t *t, char current) {
+    assert( t != NULL);
+
+    unsigned c = (unsigned) current;
+
+    if (t->children[c] == NULL)
+        t->children[c] = trie_new(current);
+
+    return EXIT_SUCCESS;
+
+}
+
+/* See trie.h */
+int trie_insert_string(trie_t *t, char *word) {
+    assert(t!=NULL);
+
+    if (*word == '\0') {
+        t->is_word=1;
+        return EXIT_SUCCESS;
+
+    } else {
+
+        char curr = *word;
+
+        int rc = trie_add_node(t, curr);
+        if (rc != 0) {
```

Comment:
Let's convert `trie_add_node` to return `EXIT_SUCCESS` and `EXIT_FAILURE` as well.

Reply:
Added EXIT_FAILURE

Reply:
@liaminchi that's great but it looks like you're still comparing `rc` against `0` directly. The point of this change is to avoid things like that. You should be comparing `rc` to `EXIT_SUCCESS` or `EXIT_FAILURE`, appropriate.

Reply:
Changed!